### ISSUE TITLE
Change defaults

### ISSUE THREAD

**Issue by Code-Slave** [meeb/tubesync#28]
I cant seem to find where defaults are kept. Id like to set defaults for certain things when I add a new channel or playlist like mp4 etc etc.

also, I keep my channels seperated in subdirs (woodworking, welding, gardening) if i add that as the folder to download to, then add {source}/ to filename, the next wood working channel i add throws an error that another channel is in woodworking. basically it looks like

ytvids/woodworking/channel1
ytvids/woodworking/channel2
ytvids/gardening/channel1

my first level dir is basically a category, theis helps when i add libraries in plex or jellyfin as it set to woodworking and has all my ww channels. 

Any idea how to do similiar without having to add it in the filename field? 

Thanks

**Comment by meeb** [comment]
How many channels are you adding? Is it noticeably time-saving to have preconfigured settings groups to apply to new channels? The defaults (e.g. new channels are VP9/OPUS) are currently hard-coded, the `default=` in `sync/models.py`. I personally didn't see this as being too much of a hassle given you should only really need to add a channel once so it's not much effort to change. Again currently the format of

`/target/download/dir/video/channel-name`

(assuming a video stream) is hard coded. Adding a source option to specify a prefix group directory might be reasonably simple, but as I mentioned in another ticket I'm currently wary of the number of options being presented on the sources page so this might need to go onto a possible future "advanced" separate options form.

**Comment by Code-Slave** [comment]
I horde yt channels cause i cant stand the ads and why not :). things like yt banning hacking etc channels are the reason why. i current have archived over 200 channels.
in plex i separate the libraries based on the topics or categories (as stated above).

Something worth thinking about maybe regarding directories. the current ability to add a dir in the filename pattern and directory to download to almost seem to fight against each other. if i add {source}/ to the pattern it puts all the vids for the playlist or channel there. maybe if the first directory to download to would allow multiple channels to go into it that would solve it.  

Maybe im just not the target user as because of the quantity i need to have i higher level grouping

**Comment by meeb** [comment]
Fair enough, that sounds reasonable. I'll add custom directory for sources to the roadmap. The custom codec default idea might be added if it's sensible to stuff in somewhere. Custom directory for sources would let you add in a directory name here:

`/path/to/downloads/video/[CUSTOM DIRECTORY HERE]/channelname/file.mkv`

Which should do what you want.

**Comment by Code-Slave** [comment]
That sounds perfect and basically mimics how i currently map my dirs

**Comment by Paul5656** [comment]
I'd also like a way to permanently set defaults. As much as you don't add many sources daily it's just more convenient in the long run if nothing else.

**Comment by OmgImAlexis** [comment]
Is there anything I can do to help get this added? I'm trying to add 500+ channels and this is going to be really tiresome to do it manually one.

Would be nice for a bulk add option too. Should I open a second issue for this?

**Comment by meeb** [comment]
Yes a bulk-add feature would be a different issue, please open a new issue for that.
